File: api/OHC/Domain/Entities/Survey/Survey.cs
using Domain.Entities.Survey.Helpers;
using Domain.Enums;
using System;

namespace Domain.Entities.Survey
{
    public class Survey
    {
        public int Id { get; private set; }
        public FactorTypeEnum FactorType { get; private set; }
        public string FactorName { get; private set; }
        public DateTime Date { get; private set; }
        public string Performer { get; private set; }
        public string Position { get; private set; }
        public string Place { get; private set; }
        public string Interpretation { get; private set; }
        public string Result { get; private set; }
        public DateTime? NextSurveyDate { get; private set; }
        public bool IsSurveySuspend { get; private set; }
        public string NormValue { get; private set; }
        public string Device { get; private set; }
        public string DeviceKind { get; private set; }
        public string MarkedParameter { get; private set; }
        public string IdentificationSurveyMethod { get; private set; }
        public string DescriptionSurveyMethod { get; private set; }
        public string IntermediateZoneH { get; private set; }
        public string IntermediateZoneE { get; private set; }
        public string RiskZoneE { get; private set; }
        public string RiskZoneH { get; private set; }
        public string DengerousZoneE { get; private set; }
        public string DengerousZoneH { get; private set; }
        public string Exposure { get; private set; }

        public static Survey CreateNew(SurveyProperties surveyProperties)
        {
            Survey survey = new Survey()
            {
                FactorType = surveyProperties.FactorType,
                FactorName = surveyProperties.FactorName,
                Date = surveyProperties.Date,
                Performer = surveyProperties.Performer,
                Result = surveyProperties.Result,
                Position = surveyProperties.Position,
                Place = surveyProperties.Place,
                Interpretation = surveyProperties.Interpretation,
                NormValue = surveyProperties.NormValue,
                Device = surveyProperties.Device,
                DeviceKind = surveyProperties.DeviceKind,
                NextSurveyDate = surveyProperties.NextSurveyDate,
                IsSurveySuspend = surveyProperties.IsSurveySuspend,
                MarkedParameter = surveyProperties.MarkedParameter,
                IdentificationSurveyMethod = surveyProperties.IdentificationSurveyMethod,
                DescriptionSurveyMethod = surveyProperties.DescriptionSurveyMethod,
                IntermediateZoneE = surveyProperties.IntermediateZoneE,
                IntermediateZoneH = surveyProperties.IntermediateZoneH,
                RiskZoneE = surveyProperties.RiskZoneE,
                RiskZoneH = surveyProperties.RiskZoneH,
                DengerousZoneE = surveyProperties.DengerousZoneE,
                DengerousZoneH = surveyProperties.DengerousZoneH,
                Exposure = surveyProperties.Exposure
            };

            return survey;
        }

        public void ChangeDetails(SurveyProperties surveyProperties)
        {
            FactorType = surveyProperties.FactorType;
            FactorName = surveyProperties.FactorName;
            Date = surveyProperties.Date;
            Performer = surveyProperties.Performer;
            Result = surveyProperties.Result;
            Position = surveyProperties.Position;
            Place = surveyProperties.Place;
            Interpretation = surveyProperties.Interpretation;
            NormValue = surveyProperties.NormValue;
            Device = surveyProperties.Device;
            DeviceKind = surveyProperties.DeviceKind;
            NextSurveyDate = surveyProperties.NextSurveyDate;
            IsSurveySuspend = surveyProperties.IsSurveySuspend;
            MarkedParameter = surveyProperties.MarkedParameter;
            IdentificationSurveyMethod = surveyProperties.IdentificationSurveyMethod;
            DescriptionSurveyMethod = surveyProperties.DescriptionSurveyMethod;
            IntermediateZoneE = surveyProperties.IntermediateZoneE;
            IntermediateZoneH = surveyProperties.IntermediateZoneH;
            RiskZoneE = surveyProperties.RiskZoneE;
            RiskZoneH = surveyProperties.RiskZoneH;
            DengerousZoneE = surveyProperties.DengerousZoneE;
            DengerousZoneH = surveyProperties.DengerousZoneH;
            Exposure = surveyProperties.Exposure;
        }        
    }
}

File: ui/src/app/modules/survey/interfaces/survey.ts
export class Survey {
  id?: number;
  type_key: string = "";
  name: string = "";
  date: string | Date = "";
  nextSurveyDate: string | Date = null;
  isSurveySuspend: boolean = false;

  position?: string;
}

export interface ISurveyForm {
  id?: number;
  value: string;
  date?: string | Date;
  nextSurveyDate?: string | Date;
}

export interface IHashSurveys {
  [key: string]: Survey[];
}

export const SURVEY_TYPES_PROPERTIES: { [key: string]: string[] } = {
  Dust: [
    "factorName",
    "date",
    "place",
    "performer",
    "result",
    "normValue",
    "interpretation",
    "deviceKind",
    "position",
    "nextSurveyDate",
    "isNormExceeded",
    "isSurveySuspend",
    "isSurveyUsed"
  ],
  Chemical: [
    "factorName",
    "date",
    "place",
    "performer",
    "result",
    "interpretation",
    "deviceKind",
    "position",
    "nextSurveyDate",
    "isSurveySuspend",
    "isSurveyUsed"
  ],
  Biological: [
    "factorName",
    "date",
    "place",
    "markedParameter",
    "identificationSurveyMethod",
    "result",
    "descriptionSurveyMethod",
    "deviceKind",
    "position",
    "nextSurveyDate",
    "isSurveySuspend",
    "isSurveyUsed"
  ],
  Physical: [
    "factorName",
    "date",
    "place",
    "performer",
    "result",
    "normValue",
    "interpretation",
    "deviceKind",
    "position",
    "nextSurveyDate",
    "isNormExceeded",
    "isSurveySuspend",
    "isSurveyUsed"
  ],
  Lighting: [
    "factorName",
    "date",
    "place",
    "performer",
    "result",
    "interpretation",
    "deviceKind",
    "position",
    "nextSurveyDate",
    "isSurveySuspend",
    "isSurveyUsed"
  ],
  Other: ["factorName", "position", "isSurveyUsed"],
  ElectromagneticField_LIST_HEADER: [
    "factorName",
    "date",
    "place",
    "performer",

    "intermediate_zone",

    "risk_zone",

    "danger_zone",

    "exposure",
    "device",
    "deviceKind",
    "position",
    "nextSurveyDate",
    "isSurveySuspend",
    "isSurveyUsed"
  ],
  ElectromagneticField_LIST_BODY: [
    "factorName",
    "date",
    "place",
    "performer",

    "intermediateZoneE",
    "intermediateZoneH",

    "riskZoneE",
    "riskZoneH",

    "dengerousZoneE",
    "dengerousZoneH",

    "exposure",
    "device",
    "deviceKind",
    "position",
    "nextSurveyDate",
    "isSurveySuspend",
    "isSurveyUsed"
  ],
  ElectromagneticField_FORM: [
    "factorName",
    "date",
    "place",
    "performer",
    "interpretation",
    "result",
    "position",

    "exposure",
    "device",
    "deviceKind",
    "nextSurveyDate",
    "isSurveySuspend",
    "isSurveyUsed"
  ]
};

export const SURVEY_TYPES_PROPERTIES_TRANSLATIONS: { [key: string]: string } = {
  factorName: "Nazwa czynnika",
  date: "Data pomiaru",
  place: "Miejsce pomiaru",
  performer: "Wykonujący pomiar",
  result: "Wynik pomiaru",
  normValue: "Norma",
  interpretation: "Interpretacja pomiaru",
  position: "Stanowisko",
  nextSurveyDate: "Kolejne badanie",

  placeDetail: "Miejsce pomiaru czynnika",

  sample: "Próbka",
  markedParameter: "Oznaczany parametr",
  identificationSurveyMethod: "Identyfikacja metody badawaczej",
  descriptionSurveyMethod: "Opis metody badawczej",

  intermediate_zone: "Strefa pośrednia",
  intermediateZoneE: "E",
  intermediateZoneH: "H",
  risk_zone: "Strefa zagrożenia",
  riskZoneE: "E",
  riskZoneH: "H",
  danger_zone: "Strefa niebezpieczna",
  dengerousZoneE: "E",
  dengerousZoneH: "H",

  isNormExceeded: "Czy przekroczono normę?",
  isSurveySuspend: "Czy zaprzestano badań?",
  isSurveyUsed: "Czy użyć tego badania?",

  exposure: "Narażenie",
  device: "Urządzenie",
  deviceKind: "Typ urządzenie"
};

export const SURVEY_TYPES_NAMES: { [key: string]: string } = {
  Dust: "Pyły",
  Chemical: "Czynniki chemiczne",
  Biological: "Czynniki biologiczne",
  Physical: "Czynniki fizyczne",
  Lighting: "Oświetlenie czynnik uciążliwy",
  ElectromagneticField: "Pole elektromagnetyczne",
  Other: "Inne czynniki niebezpieczne"
};
